Doc/tools/extensions/c_annotations.py

```diff
@@ -265,6 +265,36 @@ def run(self) -> list[nodes.Node]:
         return [node]
 
 
+class CAPIToolsBanner(SphinxDirective):
+    """CAPI Tools banner."""
```

Comment:
From an implementation perspective, it would be better to use an `.. include::` directive rather than adding custom code here.

I don't have a strong opinion on the banner in general, but I wonder if adding it to every page is a bit much? I defer to the C API group for deliberations here, though.

A

Reply:
We've had the existing note in the extending/embedding intro for a long time, but it doesn't really work for anyone coming in from a deep link (including search results). A banner seems like a plausible improvement.

Given the possibility of anchor targets dropping people in well down the pages, though, I do wonder if a hovering banner like the ones added to finalised PEPs might be a better technical approach.